File: app/assets/javascripts/channels/smoke_binpartner_traider.js
$(function() {
  if ($('.ws_sender').attr('id') == "smoke-binpartner-trader") {
    App.smoke_binpartner_traider = App.cable.subscriptions.create({
      channel: "SmokeBinpartnerTraiderChannel"
    });

    $('.pid_process').click(function(){
      $('.stop').click()
      App.smoke_binpartner_traider.send(JSON.stringify({
        pid_cucumber: "pid_cucumber"
      }))
    });

};

    $('#smoke-binpartner-trader').click(function(){
      $('.ws label').css('color','yellow').text("Tests runs, expect...")
      $(".step_cucumber").click();
      $(".start").click();
      window.scrollTo(0, 999999);

      App.smoke_binpartner_traider.send(JSON.stringify({
        smoke_binpartner_traider: true, //!!!!!

        sleep_retried: $('#generator_sleep_retried').val(),
        tariff: $('#generator_tariff').val(),
        between: $('.count_between').val(),
        count_cashier: $('.count_cashier').val(),
        count_cashier_dep: $('.count_cashier_dep').val(),
        count_payouts: $('.count_payouts').val(),
        count_payouts_sum: $('.count_payouts_sum').val(),
        count_deals: $('.count_deals').val(),
        count_deals_sum: $('.count_deals_sum').val(),
        count_pay_approve: $('.count_pay_approve').val(),
        bonus_dep: $("#generator_scenario_cancel_bonus").prop('checked'),
        turbo: $("#generator_scenario_turbo").prop('checked'),
        binary: $("#generator_scenario_binary").prop('checked'),
        count_features: get_data(),
        refcode: $('.refcode').val(),
        thread: $('#generator_thread').val(),
        yopmail : $('#generator_scenario_yopmail').prop('checked'),
        clear_cache: $('#generator_clear_clear_cache').prop('checked'),
        plus_rspec: $('#generator_plus_rspec').prop('checked'),
        geo: $('#generator_geo').val(),
        proxy_server: $('#generator_proxy_server').val(),
        proxy_http: "--proxy-server=" + $('#generator_proxy_http').val(),
        ip_address: App.config.ip,
        param_headless: $('#generator_param_headless').val(),

      }));

      notify_status_testing();
      return false;
  })
});

File: app/channels/ws/binpartner/smoke_binpartner_traider.rb
require "./features/helpers/decor"

module Server
  class SmokeBinpartnerTraider
    attr_accessor :smoke_binpartner_traider, :count_features,
                  :yopmail, :sleep_retried, :refcode, :tariff,

                  :count_cashier, :count_cashier_dep, :bonus_dep,
                  :count_payouts, :count_payouts_sum,
                  :count_deals, :count_deals_sum, :between, :turbo, :binary,
                  :count_pay_approve,

                  :thread, :geo, :clear_cache, :proxy_http,
                  :proxy_server, :plus_rspec, :clear_cache, :refcode, :ip_address, :param_headless

    def initialize(params)
      @params = params

      params.each { |key, value| public_send("#{key}=", value) }
    end

    def get_link
      JSON.parse(@params.to_json).map { |key, value| key == "count_features" ? value.to_s : "#{key}=#{value.to_s}" }.join(" ")
    end

    def message
      thread.to_i.times do
        if clear_cache == "true"
          puts "Clear cache".yellow
          system "rake tmp:cache:clear"
          system "rm -rf tmp"
          puts "Cache clear done".yellow
        end
        puts
        puts get_link.yellow
        puts;
        send_broadcast "smoke_binomo", "Run Smoke test Binpartner-traider"
        system get_link
        send_broadcast "smoke_binomo", "Done Smoke test Binpartner-traider"

      end
    end

  end
end
